### PR TITLE
SK-785, SK-819: Bump @kiltprotocol/* from 0.25.5 to 0.26.0

### PR DESCRIPTION
**Pull request by arty-name**
SK-785, SK-819

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,16 +24,16 @@
...
-    "@kiltprotocol/core": "^0.25.5",
-    "@kiltprotocol/did": "^0.25.5",
-    "@kiltprotocol/messaging": "^0.25.5",
-    "@kiltprotocol/types": "^0.25.5",
-    "@kiltprotocol/utils": "^0.25.5",
-    "@kiltprotocol/vc-export": "^0.25.5",
-    "@polkadot/keyring": "^8.2.2",
-    "@polkadot/util": "^8.2.2",
-    "@polkadot/util-crypto": "^8.2.2",
+    "@kiltprotocol/chain-helpers": "0.26.0-beta.3",
```
Comment (Diiaablo95) [inline package.json:27]: Could be `0.26.0-rc.4` now 😁

File: src/backend/endpoints/session.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ async function handler(
     data: Crypto.coToUInt8(encryptedChallenge),
     nonce: Crypto.coToUInt8(nonce),
     publicKey: keyAgreement.publicKey,
-    peerPublicKey: Crypto.coToUInt8(encryptionKey.publicKeyHex),
+    peerPublicKey: Crypto.coToUInt8(encryptionKey.publicKey),
```
Comment (Diiaablo95) [inline src/backend/endpoints/session.ts:63]: `encryptionKey.publicKey` is already a `UInt8Array`.

File: src/backend/utilities/encryptMessage.ts [hunk 3]
```diff
@@ -1,36 +1,31 @@
...
-import { DefaultResolver } from '@kiltprotocol/did';
 
 import { fullDidPromise } from './fullDid';
 import { encryptionKeystore } from './keystores';
 import { configuration } from './configuration';
 
 export async function encryptMessage(
   message: Message,
-  encryptionKeyId: IDidKeyDetails['id'],
+  encryptionKeyId: DidKey['id'],
```
Comment (Diiaablo95) [inline src/backend/utilities/encryptMessage.ts:10]: If this is the full key URI, then it's a DidPublicKey['id']. If it's just the ID, then `message.encrypt()` below must use the full URI and not just the ID. If the tests work, I assume the type annotation here is not correct.

File: src/backend/utilities/encryptMessage.ts [hunk 4]
```diff
@@ -1,36 +1,31 @@
...
+    encryptionKey.id,
+    fullDid,
     encryptionKeystore,
+    encryptionKeyId,
   );
 }
 
 export async function encryptMessageBody(
-  encryptionKeyId: IDidKeyDetails['id'],
+  encryptionKeyId: DidKey['id'],
```
Comment (Diiaablo95) [inline src/backend/utilities/encryptMessage.ts:28]: Same here.

File: src/backend/utilities/fullDid.ts [hunk 5]
```diff
@@ -23,50 +29,51 @@ import { logger } from './logger';
...
-      [keyAgreement]: didDetails.getKeyIds(keyAgreement),
-    },
-    lastTxIndex: await DidChain.queryLastTxCounter(didDetails.did),
-  });
-}
-
 async function compareKeys(
   derived: KeyringPair | Keypair,
-  fullDid: FullDidDetails,
+  resolved: DidVerificationKey | DidEncryptionKey | undefined,
```
Comment (Diiaablo95) [inline src/backend/utilities/fullDid.ts:69]: `DidVerificationKey | DidEncryptionKey` can simply be `DidKey`.

File: src/backend/utilities/sessionStorage.ts [hunk 6]
```diff
@@ -12,7 +12,7 @@ import { Attestation } from '@kiltprotocol/core';
 export interface Session {
   sessionId: string;
   did?: IDidDetails['did'];
-  encryptionKeyId?: IDidKeyDetails['id'];
+  encryptionKeyId?: DidKey['id'];
```
Comment (Diiaablo95) [inline src/backend/utilities/sessionStorage.ts:15]: This would also be `DidPublicKey['id']`.

File: src/backend/utilities/sessionStorage.ts [hunk 7]
```diff
@@ -24,7 +24,7 @@ export interface Session {
 
 export type SessionWithDid = Session & {
   did: IDidDetails['did'];
-  encryptionKeyId: IDidKeyDetails['id'];
+  encryptionKeyId: DidKey['id'];
```
Comment (Diiaablo95) [inline src/backend/utilities/sessionStorage.ts:27]: Same as above.

File: src/frontend/utilities/session.ts [hunk 8]
```diff
@@ -11,15 +11,15 @@ interface PubSubSession {
   ) => Promise<void>;
   close: () => Promise<void>;
   send: (message: IEncryptedMessage) => Promise<void>;
-  encryptionKeyId: IDidKeyDetails['id'];
+  encryptionKeyId: DidKey['id'];
```
Comment (Diiaablo95) [inline src/frontend/utilities/session.ts:14]: Also here.

File: src/frontend/utilities/session.ts [hunk 9]
```diff
@@ -11,15 +11,15 @@ interface PubSubSession {
...
   encryptedChallenge: string;
   nonce: string;
 }
 
 interface InjectedWindowProvider {
   startSession: (
     dAppName: string,
-    dAppEncryptionKeyId: IDidKeyDetails['id'],
+    dAppEncryptionKeyId: DidKey['id'],
```
Comment (Diiaablo95) [inline src/frontend/utilities/session.ts:22]: And here.

**Review by rflechtner — COMMENTED** [review]
Looks good to me so far!

**Review by lukeg90 — APPROVED** [review]

**Comment by rflechtner** [comment]
Remember when we talked about deduplication with yarn? I wasn't aware that yarn v1 does not let you run `yarn dedupe` and instead lies to you about doing deduplication implicitly. But try running `npx yarn-deduplicate yarn.lock` instead, that should clean up a bit and may reduce bundle size? Same probably goes for sporran.

**Pull request merged**